src/ucp/api/ucp.h

```diff
@@ -2177,8 +2177,105 @@ ucs_status_t ucp_worker_fence(ucp_worker_h worker);
  */
 ucs_status_t ucp_worker_flush(ucp_worker_h worker);
 
+/**
+ * @ingroup UCP_COMM
+ * @brief Atomic operation requested for ucp_atomic_post
+ *
+ * This enumeration defines which atomic memory operation should be performed
+ * by the ucp_atomic_post family of fuctions. All of these are send only
+ * and will not give back a request handle.
+ */
+typedef enum {
+    UCP_ATOMIC_POST_OP_ADD, /* Atomic add */
+    UCP_ATOMIC_POST_OP_LAST,
+} ucp_atomic_post_op_t;
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Atomic operation requested for ucp_atomic_fetch
+ *
+ * This enumeration defines which atomic memory operation should be performed
+ * by the ucp_atomic_fetch family of functions. All of these functions
+ * will fetch data from the remote node
+ */
+typedef enum {
+    UCP_ATOMIC_FETCH_OP_FADD, /* Atomic Fetch and add */
+    UCP_ATOMIC_FETCH_OP_SWAP, /* Atomic swap */
+    UCP_ATOMIC_FETCH_OP_CSWAP, /* Atomic conditional swap */
+    UCP_ATOMIC_FETCH_OP_LAST,
+} ucp_atomic_fetch_op_t;
 
 /**
+ * @ingroup UCP_COMM
+ * @brief Post an atomic memory operation.
+ *
+ * This routine will post an atomic memory operation to remote memory.
+ * Return from the function does not guarantee any completion. A user must
+ * call @ref ucp_ep_flush to guarantee local and remote completion.
+ *
+ * @param [in] ep          UCP endpoint.
+ * @param [in] opcode      One of @ref ucp_atomic_post_op_t.
+ * @param [in] value       Source operand for AMO.
+ * @param [in] op_size     Size of value in bytes
+ * @param [in] remote_addr Remote address to operate on.
+ * @param [in] rkey        Remote key handle for the remote address.
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_atomic_post(ucp_ep_h ep, ucp_atomic_post_op_t opcode, uint64_t value,
+                             size_t op_size, uint64_t remote_addr, ucp_rkey_h rkey);
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Post an atomic fetch operation.
+ *
+ * This routine will post an atomic fetch operation to remote memory.
```

Comment:
We are not very consistent in terminology here . In some. Cases you use AMO and in another ״atomic operation ".